### ISSUE TITLE
Bug when compiling for Windows from Linux

### ISSUE THREAD

**Issue by DarsFabian** [golang/go#49576]
### What version of Go are you using (`go version`)?

<pre>
$ go version
go1.17.2 linux/amd64
</pre>

### What operating system and processor architecture are you using (`go env`)?

<details><summary><code>go env</code> Output</summary><br><pre>
$ go env
GO111MODULE=""
GOARCH="amd64"
GOBIN=""
GOCACHE="/home/dars_fabian/.cache/go-build"
GOENV="/home/dars_fabian/.config/go/env"
GOEXE=""
GOEXPERIMENT=""
GOFLAGS=""
GOHOSTARCH="amd64"
GOHOSTOS="linux"
GOINSECURE=""
GOMODCACHE="/home/dars_fabian/go/pkg/mod"
GONOPROXY=""
GONOSUMDB=""
GOOS="linux"
GOPATH="/home/dars_fabian/go"
GOPRIVATE=""
GOPROXY="https://proxy.golang.org,direct"
GOROOT="/usr/local/go"
GOSUMDB="sum.golang.org"
GOTMPDIR=""
GOTOOLDIR="/usr/local/go/pkg/tool/linux_amd64"
GOVCS=""
GOVERSION="go1.17.2"
GCCGO="gccgo"
AR="ar"
CC="gcc"
CXX="g++"
CGO_ENABLED="1"
GOMOD="/dev/null"
CGO_CFLAGS="-g -O2"
CGO_CPPFLAGS=""
CGO_CXXFLAGS="-g -O2"
CGO_FFLAGS="-g -O2"
CGO_LDFLAGS="-g -O2"
PKG_CONFIG="pkg-config"
GOGCCFLAGS="-fPIC -m64 -pthread -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build3135623564=/tmp/go-build -gno-record-gcc-switches"
</pre></details>

### What did you do?
I tried to create a simple executable file from Linux that runs on Windows. It runs perfectly fine on Linux but then when I compile it for Windows it gets blocked by the antivirus but that is not really the issue. Windows interprets it as a 16-bits executable file instead of a 64-bits executable and I didn't find any answer to my issue, it seems like it is a bug.
`GOOS=windows GOARCH=amd64 go build -o file.exe main.go`

Here is my code:
https://play.golang.org/p/SQXm7-gfBii

### What did you expect to see?

I expected to see "Frog" printed on my screen.

### What did you see instead?

Application 16 bits non prise en charge

Impossible de démarrer ou d'exécuter le programme ou la fonction
«\??/C:\Users\fabiaDesktop\file.exe » en raison d'une
incompatibilité avec les versions 64 bits de Windows. Contactez
l'éditeur du logiciel pour demander si une version compatible avec
Windows 64 bits est disponible.

**Comment by seankhliao** [comment]
Duplicate of #44328

Doesn't look like a Go bug to us
